crypto/rand/randfile.c

```diff
@@ -208,8 +208,12 @@ int RAND_write_file(const char *file)
          * should be restrictive from the start
          */
         int fd = open(file, O_WRONLY | O_CREAT | O_BINARY, 0600);
-        if (fd != -1)
+
+        if (fd != -1) {
             out = fdopen(fd, "wb");
+            if (out == NULL)
+                close(fd);
```

Comment:
Do we want to add a call to `ERR_raise_data()` as is done in #25081?

NOTE: Whichever of these two PRs is merged, the other should be closed

Reply:
Note [@IDisposable's comment](https://github.com/openssl/openssl/issues/25064#issuecomment-2291842888) that #25081 "changes the flow so we don't even attempt the subsequent attempt to use openssl_fopen in write+binary mode"